### ISSUE TITLE
[Bug] multiple Discord clients breaking Music Presence

### ISSUE THREAD

**Issue by moeshpit** [ungive/discord-music-presence#61]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

title explains it

### Reproduction Steps

1. have multiple clients open at the same time (stable, ptb, canary)
2. profit

### Expected behavior

that it 
a) worked on both instances
b) worked on the one that didn't disable rich presence's in general

### Device Info

- Discord Music Presence version: 2.2.4
- Operating system: Windows 10
- The media player you used (if applicable): foobar2000
- The Discord client I used at the same time: stable, PTB

**Comment by ungive** [comment]
I was able to reproduce this behaviour using Discord stable and PTB, each logged in with different user accounts. Here is what I found:

- You can have control over where it shows by starting the instance of Discord where you want it to show first, while the others are closed, then starting the other instance
- On Windows, Discord does create two named pipes for Discord RPC/Music Presence to connect to, one for each Discord instance, e.g. `\\.\\pipe\\discord-ipc-0` and `\\.\\pipe\\discord-ipc-1`, which makes it appear it would be possible to select which Discord instance to connect to. You can confirm this by running this command in a Powershell window while multiple Discord instances are opened: `[System.IO.Directory]::GetFiles("\\.\\pipe\\") | Select-String -Pattern '\\discord-'`
- The discord-rpc library only connects to the first pipe that is created, not considering other ones: https://github.com/ungive/discord-rpc/blob/5a24d2d24e6aa16b6b85dbb7010cd0e6015d1d1c/src/connection_win.cpp#L35
- I've modified this line of code while two instances were opened, but when I log which user is connected and test the presence, the status is still shown for only one user, not the other. So `ipc-0` and `ipc-1` both control the status for the same user. Also, if e.g. Discord instance A (ipc-0, PTB) was started first and instance B (ipc-1, stable) was started second and both ipc-0 and ipc-1 set the status for instance A, closing instance A (and therefore ipc-0, which leaves ipc-1) then that all of a sudden makes ipc-1 set the status for instance B. Starting A again creates ipc-0 again and makes both ipc-0 and ipc-1 set the status for instance A again
- In short: ipc-1 seems to always be "redirected" to ipc-0, unless ipc-0 does not exist
- I've additionally installed the Canary version of Discord and ran stable, PTB, Canary (in that order) and all pipes (ipc 0, 1 and 2) all set the status for the stable instance of Discord (ipc-0)
- The behaviour is the same even when using stable+PTB and not using stable at all

So in conclusion there is no way to select which Discord instance Music Presence connects to, it always connects to one and no matter which pipe you connect to, it always shows the status in the first instance that was started. The only way to select the Discord instance in which your status is shown is by starting that instance first.

I hope that helps!
